build.sbt

```diff
@@ -246,7 +249,9 @@ lazy val commonSettings = Seq(
   sourceManaged := baseDirectory.value / "src_managed",
   resourceManaged := baseDirectory.value / "resource_managed",
   libraryDependencies ++= Dependencies.common,
-  testOptions += Tests.Argument(TestFrameworks.JUnit, "-q", "--verbosity=1")
+  testOptions += Tests.Argument(TestFrameworks.JUnit, "-q", "--verbosity=1"),
+  Compile / packageDoc / publishArtifact := false,
+  Dependencies.genjavadocVersion
```

Comment:
This now results in a bunch of warnings like this:
```
[warn] there are 20 keys that are not used by any other settings/tasks:
[warn]  
[warn] * daffodil / unidocGenjavadocVersion
[warn]   +- /home/runner/work/daffodil/daffodil/project/Dependencies.scala:73
...
```

I guess my idea of adding this globally wasn't a great idea. Maybe we just add
```scala
.settings(Dependencies.genjavadocVersion)
```
after we enable the JavadocPlugin for each project? Its a bit repetitive, but I personally find it a bit cleaner than the `withScalaDocConvertedToJavaDoc` function. But I don't feel strongly about it.